File: Problems_and_Solutions/day-8/Java/Solution1.java
import java.util.Arrays;
class Solution1 {
    public static void insertionSort(int arr[]) {
        for(int i=0; i<arr.length; i++){
            for(int j=i; j>0; j--){
                if(arr[j-1] > arr[j]){
                    int temp = arr[j-1];
                    arr[j-1] = arr[j];
                    arr[j] = temp;
                }
            }
        }
    }

    public static void main(String[] args){
        int a[] ={4, 1, 3, 9, 7};
        insertionSort(a);
        System.out.println(Arrays.toString(a));
    }
}

File: Problems_and_Solutions/day-8/cpp/solution 1.cpp
class Solution {
  public:
    // Please change the array in-place
    void insertionSort(vector<int>& arr) {
        // code here
        int i=0,j=0,temp=0;
        
        for(i=1;i<arr.size();i++)
        {
            temp=arr[i];
            for(j=i-1;j>=0;j--)
            {
                if(temp<arr[j])
                    arr[j+1]=arr[j];
                else
                    break;
            }
            arr[j+1]=temp;
        }
    }
};